csbmysql/resource_binding_user.go

```diff
@@ -106,8 +113,16 @@ func resourceBindingUserCreate(ctx context.Context, d *schema.ResourceData, m an
 		}
 	}
 
-	grantStatement := fmt.Sprintf("GRANT ALL ON %s.* TO %s@%s", quotedIdentifier(cf.database),
-		quotedIdentifier(username), quotedIdentifier(bindingUserHostAll))
+	permission := "ALL"
+	if readOnly {
+		permission = "SELECT"
```

Comment:
thoughts: For MySQL read-only permissions, there are actually several privileges that are typically granted

- SELECT - for reading data
- SHOW VIEW - for viewing table structures
- SHOW DATABASES - for listing databases
- PROCESS - for viewing server status
I will discard the last two because we only create one database, and the last one is too advanced. I have doubts about the second one, although I wouldn't include it in this first approach